mrblib/aws-s3.rb

```diff
@@ -38,18 +41,17 @@ def calculate_signature(method, path, headers)
       method = method.upcase
       time = Time.now.utc
       path = "/#{path}" unless path.start_with? '/'
-      path = "/#{@bucket_name}#{path}"
 
-      headers['x-amz-content-sha256'] = Digest::SHA256.hexdigest(headers['Body']) unless
-        headers['x-amz-content-sha256']
+      headers['Host'] = @s3_endpoint.host
+      headers['x-amz-content-sha256'] = Digest::SHA256.hexdigest(headers['Body'])
       headers['x-amz-security-token'] = @security_token if @security_token
       headers['Date'] = headers['x-amz-date'] = time.strftime("%Y%m%dT%H%M%SZ")
       if method == 'PUT'
         headers['Content-Length'] = headers['Body'].bytesize
         headers['Content-Encoding'] = 'aws-chunked'
       end
       headers['Accept'] = SimpleHttp::DEFAULT_ACCEPT
-      headers['Connection'] = 'close'
+      headers['Connection'] = 'keep-alive'
```

Comment:
Is there any reason why `keep-alive` ?

Reply:
I was encountered an error when Connection header is `close`, although I am not sure why it is.

Reply:
BTW simplehttp forces `close` somehow. Maybe need fix for it too...
https://github.com/matsumotory/mruby-simplehttp/blob/de9c250eadfabf48e289cbf5039b47e12a691ccb/mrblib/simplehttp.rb#L194